### DISCUSSION TITLE
Questions about Registering Schema

### DISCUSSION THREAD

**Discussion by neapolis123** [apache/pulsar#18850] [Q&A]
Hi, I just wanted to ask about what internal state does the producer take after failing to register the Schema because its incompatible( Line 592 in ProducerImp ) . From what I have understood the state does not change but then wouldn't be logical to just close the producer. Since going back to Ready state will just result in a deadlock of an endless loop of trying to register the schema.

I hope this is the right place to ask such a question.

**Comment by sijie** [comment]
If the schema is incompatible, creating a producer should fail. Did you see any problems?

**Comment by neapolis123** [comment]
I am writing my thesis about Pulsar and a part of it is to model producers and consumers as state machines. When the reading the code I couldn't figure out what state the producer assumes when going from Ready to RegisteringSchema and then the registering fails. Can you please refer to where in the source code is the state changed to "Failed" when the function tryRegisteringSchema catches an IncompatibleSchemaException in the ProducerImp class  ? 
I am sorry if this is not the appropriate place to discuss this. 
My email : spirovsk@gmail.com If you deem it better to discuss this privately.

**Comment by Technoboy-** [comment]
ClientCnx#sendGetOrCreateSchema will send the request to broker.  
ClientCnx#handleGetOrCreateSchemaResponse will handle the response.

**Comment by neapolis123** [comment]
> ClientCnx#sendGetOrCreateSchema will send the request to broker.
> ClientCnx#handleGetOrCreateSchemaResponse will handle the response.

So what does ProducerImpl#tryRegistringSchema do ?
